Support Paper 1.20.1

muqhc opened the pull request:
## 개요
Paper 1.20.1를 지원합니다.

## 주요 변경사항

### 요약

---
- Paper 1.20.1 에서 사용 가능하도록 종속성 업데이트.
- 변경된 최신 tap-api 와의 호환을 위한 일부 psychics-core 및 능력자 코드 수정.
- 제대로 된 abilities.zip 생성.
---

### 구체적인 변경사항

- Paper 1.17.1 -> 1.20.1
- 종속성 업데이트.
- - io.github.monun:tap-api:4.1.9 -> io.github.monun:tap-api:4.9.8
- - io.github.monun:kommand-api:2.6.6 -> io.github.monun:kommand-api:3.1.7
- - io.github.monun:invfx-api:3.0.1 -> io.github.monun:invfx-api:3.3.2
- 호환을 위한 psychics-core 및 능력자 수정.
- - [x] FakeEntity<T:Entity> 제너릭 문제 해결.
- - [x] tap-api 에 폭죽 생성 메소드가 빠져 따로 구현 해야됨.
- 기타 업데이트
- - Gradle 7.2 -> 7.6
- - Kotlin 1.5.21 -> 1.7.21
- - Java 16 -> 17
- 제대로 된 abilities.zip 생성

muqhc committed:
Update gradle-wrapper.properties

muqhc committed:
Update build.gradle.kts

muqhc committed:
Update build.gradle.kts

muqhc committed:
Update README.md

muqhc committed:
resolve libs and optimize task

muqhc committed:
solve FakeEntity generic problem

muqhc committed:
update README.md

muqhc committed:
Complete Migration to paper 1.19

muqhc committed:
solve last FakeEntity generic problem

muqhc committed:
update plugin.yml

muqhc committed:
update plugin.yml

muqhc committed:
fix plugin.yml

muqhc committed:
auto generate abilities.zip

kdh8219 commented:
1.19에서는 잘 작동하나, 1.19.1에서는 오류가 발생합니다!
오류 메시지가 kommand 관련된것 같네요

muqhc commented:
@kdh8219 
> 1.19에서는 잘 작동하나, 1.19.1에서는 오류가 발생합니다! 오류 메시지가 kommand 관련된것 같네요

(2022년 7월 30일 기준)
현재 tap, kommand 는 1.19.1 를 위한 구현이 없습니다.
즉, api가 v1_19_1 구현을 로드하지 못 하므로,
psychics 또한 1.19.1 에서 사용 불가 합니다.

(+ [2022-08-04] 지금은 있어서 1.19.1로 업뎃하였습니다. )

ghost commented:
Kommand 오늘 플리퀘했어요! 어서 제 PR과 이 PR이 받아져서 어르신이 다시 일주일 1능력을 실천하시면 좋겠네요 🤣🤣🤣

+ Tap dongle 1.19.1 업데이트했습니다

ghost commented:
- [Tap: 49ad3b2](https://github.com/monun/tap/commit/49ad3b2a89a93fa75298946ab6655c1edb85d81c)
- [Kommand: 6d89b69](https://github.com/monun/kommand/commit/6d89b692ba5cbb0429a6baa92b04a38176f73be1)

dytroc commented:
어제 방송에서 말하시길 구조 갈아엎는다네요!

ghost commented:
버린게 아니었어

muqhc committed:
support Paper 1.19.1

muqhc commented:
1.19.1 로 업데이트!

muqhc committed:
cleanup abilities.zip process code

muqhc committed:
make dependcies and tool version up-to-date

muqhc commented:
1.19.3

muqhc committed:
support paper 1.19.4

muqhc commented:
1.19.4

highright1234 commented:
ㄷㄷ.........

muqhc committed:
fix bugs
- fix permission require caused by kommand-api version up
- fix abilities.zip generating

muqhc committed:
1.20.1
Paper 1.19.4 -> 1.20.1
Tap 4.9.4 -> 4.9.6
Kommand 3.1.3 -> 3.1.6

muqhc commented:
1.20.1

muqhc committed:
update dependencies
tap 4.9.6 -> 4.9.8
kommand 3.1.6 -> 3.1.7
invfx 3.3.0 -> 3.3.2

muqhc committed:
support firework effect

muqhc committed:
fix syntax error FireworkSupport.kt

muqhc committed:
fix reference error

muqhc committed:
Merge pull request #7 from muqhc/patch/firework-support

Patch/firework support

muqhc commented:
24.02.19 폭죽 구현됨

muqhc commented:
tap 라이브러리 문제로 1.20.1에서 못올라 가는 중
